src/core/net/srp_server.cpp

```diff
@@ -449,14 +449,14 @@ void Server::CommitSrpUpdate(Error                    aError,
     uint32_t grantedTtl;
     bool     shouldFreeHost = true;
 
-    SuccessOrExit(aError);
-
     hostLease       = aHost.GetLease();
     hostKeyLease    = aHost.GetKeyLease();
     grantedLease    = aLeaseConfig.GrantLease(hostLease);
     grantedKeyLease = aLeaseConfig.GrantKeyLease(hostKeyLease);
     grantedTtl      = aTtlConfig.GrantTtl(grantedLease, aHost.GetTtl());
 
+    SuccessOrExit(aError);
```

Comment:
Do we need to change this? I prefer if we do not.

At `exit` label, `grantedLease` is only used when `aError == kErrorNone`.

Reply:
I see. I agree maybe initing the variables to zero instead would be better?
Thanks for fixing this.

Reply:
Sure, makes sense. I will update the commit and repush. Thanks for the feedback !